src/Common/ProxyConfiguration.h

```diff
@@ -49,6 +49,7 @@ struct ProxyConfiguration
     uint16_t port = 0;
     bool tunneling = false;
     Protocol original_request_protocol = Protocol::HTTP;
+    std::string no_proxy_hosts = "";
```

Comment:
This list is constructed from env. It is immutable for the rest of the program live.

Why we copy it to each connection?
Why do we leave it in raw format and parse each time?

Reply:
The last commit is a possible implementation of the above idea, please take a look: https://github.com/ClickHouse/ClickHouse/pull/63314/commits/3ae8176a6d094be53ebd82d2fc7b5e1c75ceccd2.